tests/orchestrator/test_call_http.py

```diff
@@ -174,3 +175,32 @@ def test_post_completed_state():
     # assert_valid_schema(result)
     assert_orchestration_state_equals(expected, result)
     validate_result_http_request(result)
+
+@pytest.mark.parametrize("content, expected_content", [
+    (None, None),
+    ("string data", "string data"),
+    ('{"key": "value"}', '{"key": "value"}'),
+    ('["list", "content"]', '["list", "content"]'),
+    ('[]', '[]'),
+    ('42', '42'),
+    ('true', 'true'),  
+    # Cases that test actual behavior (not strictly adhering to Optional[str])
```

Comment:
Although we define that the `content` should be of type `string` or `None`, we do not throw an error in the `call_http` method if `content` is of a different type. For this reason, I added these test cases, which I believe align with the request outlined by the customer in the GitHub issue.

Reply:
Oh, I didn't realize that `content` was supposed to be `string` or `None`. It seems like we probably never should have been serializing to JSON to begin with. :(

Reply:
Yeah that's what I am confused about haha.  From my understanding, we shouldn’t serialize the content into a JSON string if it’s already a string. However, our current logic seems to be doing the opposite. Not sure if @davidmrdavid  can provide more context here

Reply:
I think adding these tests is the right thing to do. Maybe we can consider updating the method's type signature, but I think that's a non-blocking decision that we can defer.

Reply:
I no longer recall the context on why the code works this way (some of it predates me), but it's clearly a bug nonetheless.

I agree the types need revising, but for now I think the restrictive type signature is probably for the better, it constraints the valid uses.

+1 to adding these tests. @nytian, can you please also add a tests for when `is_raw_string` is set to `True`? It would be ideal to cover both scenarios